src/command_modules/azure-cli-cognitiveservices/azure/cli/command_modules/cognitiveservices/_params.py

```diff
@@ -69,13 +74,64 @@ def _is_suspected_json(string):
 )
 
 
+def _sku_filter(cmd, namespace):
+    """
+    Get a list of ResourceSku and filter by existing conditions: 'kind', 'location' and 'sku_name'
+    """
+    kind = getattr(namespace, 'kind', None)
+    location = getattr(namespace, 'location', None)
+    sku_name = getattr(namespace, 'sku_name', None)
+
+    def _filter_sku(_sku):
+        if sku_name is not None:
+            if _sku.name != sku_name:
+                return False
+        if kind is not None:
+            if _sku.kind != kind:
+                return False
+        if location is not None:
+            if location.lower() not in [x.lower() for x in _sku.locations]:
+                return False
+        return True
+
+    return [x for x in cf_resource_skus(cmd.cli_ctx).list() if _filter_sku(x)]
+
+
+@Completer
+def sku_name_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    names = set([x.name for x in _sku_filter(cmd, namespace)])
+    return sorted(list(names))
+
+
+@Completer
+def kind_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    kinds = set([x.kind for x in _sku_filter(cmd, namespace)])
+    return sorted(list(kinds))
+
+
+@Completer
+def location_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    names = set(sum([x.locations for x in _sku_filter(cmd, namespace)], []))
+    return [x.lower() for x in sorted(list(names))]
+
+
+def get_cognitiveservices_location_type(ctx):
+    _location_type = get_location_type(ctx)
+    _location_type.update(completer=location_completer)
+    return _location_type
```

Comment:
No need to define a new arg_type; just use `get_location_type`.

Reply:
In here I need to override the default completer from the get_location_type.
So I guess I have to define a new type.

Reply:
If you use "get_location_type" and just use the `completer` kwarg, the completer you specify will override the one in the type.

Reply:
```
def get_location_type(cli_ctx):
    location_type = CLIArgumentType(
        options_list=('--location', '-l'),
        completer=get_location_completion_list,
        type=get_location_name_type(cli_ctx),
        help="Location. You can configure the default location using `az configure --defaults location=<location>`",
        metavar='LOCATION',
        configured_default='location')
    return location_type
```

This is the definition of get_location_type(), it doesn't taking `kwarg`.
Did I miss something?

Reply:
I mean when registering the parameter:
`c.argument('location', get_location_type(self.cli_ctx), completer=<>)`

Reply:
cool, change made as suggested.